src/test/regress/expected/multi_create_table.out

```diff
@@ -873,5 +873,30 @@ SELECT count(*) FROM tt1;
      6
 (1 row)
 
+BEGIN;
+CREATE SCHEMA sc;
+CREATE TABLE sc.ref(a int);
+insert into sc.ref SELECT s FROM generate_series(0, 100) s;
+SELECT create_reference_table('sc.ref');
+NOTICE:  Copying data from local table...
+ create_reference_table 
+------------------------
+ 
+(1 row)
+
+CREATE TABLE sc.hash(a int);
```

Comment:
Does it make sense to put this in a separate transaction?

Reply:
Oh, and separate schema names :)

Reply:
Hmm, I'm a bit confused by your comment. My goal here is the following:

1. `create_reference_table()` creates `sc` on all nodes, in a separate transaction
2. `create_reference_table()` creates placements and just before that issues create schema command on the node
3. `create_distributed_table()` creates (if not exists) `sc` on all nodes, in a separate transaction
4. `create_distributed_table()` creates shard placements and just before that issues create schema command on the node

This patch adds the item `1` since otherwise `2` and `3` deadlocks -- (since 2 happens in the coordinated tx whereas 3 happens seperately). I think we already have a bunch of other tests that tests the creation of the schemas on separate transactions.

(We also already have the tests in the opposite order, but I could add here for completeness if you want)

Reply:
My thinking was: if we accidentally remove `EnsureSchemaExistsOnAllNodes` in `create_distributed_table` then this test won't catch that because the call to `create_reference_table` already makes sure that the schema exists. For that we should have a `BEGIN; CREATE SCHEMA ...; SELECT create_distributed_table(...); END;` test.

You're right that without the call to `create_distributed_table` we wouldn't notice if the `EnsureSchemaExistsOnAllNodes` in `create_reference_table` had gone missing. I missed that, since it's rather subtle that this doesn't become a problem unless you do a `create_distributed_table`. It might be worth adding a comment on why we're calling `create_distributed_table`.

Reply:
> It might be worth adding a comment on why we're calling create_distributed_table.

Makes sense, I'll add the comment